packages/assets-controllers/src/NftController.ts

```diff
@@ -553,11 +593,19 @@ export class NftController extends BaseController<NftConfig, NftState> {
     const blockchainContractData: Partial<ApiNftContract> &
       Pick<ApiNftContract, 'address'> &
       Pick<ApiNftContract, 'collection'> = await safelyExecute(async () => {
-      return await this.getNftContractInformationFromContract(contractAddress);
+      return await this.getNftContractInformationFromContract(
+        contractAddress,
+        networkClientId,
+      );
     });
 
+    let { chainId } = this.config;
+    if (networkClientId) {
+      chainId =
+        this.getNetworkClientById(networkClientId).configuration.chainId;
```

Comment:
use the this.getCorrectChainId helper here?

Reply:
done here: https://github.com/MetaMask/core/pull/1698/commits/5fd50ec7deb2b8e06f3221c78abd7363df0d7039